lib/propolis/src/instance.rs

```diff
@@ -47,7 +47,14 @@ impl Drop for Instance {
     fn drop(&mut self) {
         // Make sure the inventory is cleared, even if the consumer chose not to
         // call Instance::destroy().
-        self.0.get_mut().unwrap().inventory.clear()
+        self.0.get_mut().unwrap().inventory.clear();
+
+        // If Instance::destroy() was not called prior to drop, let's cleanup
+        // the underlying VMM resource.
+        if let Some(machine) = self.0.get_mut().unwrap().machine.take() {
+            let hdl = machine.destroy();
+            let _ = hdl.destroy();
+        }
```

Comment:
Now that vmm instances can be configured for [auto-destruct](https://www.illumos.org/issues/14817), I wonder we if should just allow that to be configured rather than the explicit destroy on drop (in case there are times when we want to bail out w/o destroying the kernel vmm state)

On the flip side, we could just make that change if/when it's needed

Reply:
For right now I think holding off on auto-destruct is preferable. This change should only result in VMs that are explicitly stopped being cleaned up. If we get into a failed state or crash, or otherwise not exit cleanly, the VmmHandle should persist which could help for investigation.

Reply:
Works for me!